test/types/string/kbrady/type-cast.chpl

```diff
@@ -0,0 +1,11 @@
+record R {
+  var q = 1;
+}
+
+var x = 10;
+var y = 1.0;
+var z = new R();
+
+writeln(x.type:string);
+writeln(y.type:string);
+writeln(z.type:string);
```

Comment:
I thoroughly expect this to work, but it might be nice to add cases like:

    writeln(int(32):string);  // write a type directly

    type t = int(32);
    writeln(t:string);